base/applications/mspaint/textedit.cpp

```diff
@@ -218,9 +218,9 @@ void CTextEditWindow::FixEditPos(LPTSTR pszOldText)
     ::GetClientRect(m_hwndParent, &rcParent);
     IntersectRect(&rc, &rcParent, &rcWnd);
 
-    InterlockedIncrement(&m_nMoveSizeLock);
+    ++m_nMoveSizeLock;
```

Comment:
interlocked or not, you still haven't told me why we need these `m_nMoveSizeLock` things.

Reply:
There has a bug in zooming the textbox.

Reply:
Manually calling `MoveWindow` will causes `WM_MOVE` and `WM_SIZE` messages.
`m_nMoveSizeLock == 0` can check whether the message is from the user or the application.

Reply:
If the message was from the user, we have to call `InvalidateEditRect();` in `OnMove` and `OnSize`.
If it was from the application, we don't have to call `InvalidateEditRect();` in `OnMove` and `OnSize`.

Reply:
`s/m_nMoveSizeLock/m_nAppIsMovingOrSizing/`